### PR TITLE
Iteration 5 tt

### PR DESCRIPTION
**Pull request by tylertomlinson**
Added methods from Iteration 5:
average_win_percentage
most_goals_score
fewest_goals_scored

-- Added to Stat Tracker ... Passes Rspec

### CODE REVIEW

File: lib/game_teams_stats.rb [hunk 1]
```diff
@@ -1,22 +1,29 @@
...
   def average_win_percentage(team_id)
-    total_wins_per_team / @game_teams_collection.total_games_per_team(team_id).to_f
+    total_wins_per_team(team_id.to_i) / @game_teams_collection.total_games_per_team(team_id.to_i).to_f
   end
 
   def total_goals_by_team_id(team_id)
     @game_teams_collection.games_by_team_id(team_id).sum {|game_team| game_team.goals.to_i}
   end
 
+  def most_goals_scored(team_id)
```
Comment (d-atkins) [inline lib/game_teams_stats.rb]: Is there a reason there are two versions of this same method?

File: lib/game_teams_stats.rb [hunk 2]
```diff
@@ -25,6 +32,14 @@ def best_offense
     @game_teams_collection.unique_team_ids.max_by {|team_id| average_goals_per_team_id(team_id)}
   end
 
+  def most_goals_scored(team_id)
```
Comment (d-atkins) [inline lib/game_teams_stats.rb:30]: ^^^

File: lib/game_teams_stats.rb [hunk 3]
```diff
@@ -7,12 +7,12 @@ def initialize(game_teams_collection)
     @game_teams_collection = game_teams_collection
   end
 
-  def total_wins_per_team
-    @game_teams_collection.game_teams_array.select {|game_team| game_team.result == "WIN"}.count
+  def total_wins_per_team(team_id)
```
Comment (Yetidancer) [inline lib/game_teams_stats.rb:10]: thanks for changing all of these

File: lib/stat_tracker.rb [hunk 4]
```diff
@@ -106,4 +106,19 @@ def best_fans
   def worst_fans
     @game_teams.worst_fans_ids.map {|id| @team_collection.team_name_by_id(id)}
   end
+
+  def most_goals_scored(team_id)
```
Comment (mronauli) [inline lib/stat_tracker.rb:110]: looks good!

File: test/game_teams_stats_test.rb [hunk 5]
```diff
@@ -18,11 +18,11 @@ def test_it_has_attributes
   end
 
   def test_it_can_get_total_wins_per_team
-    assert_equal 25, @game_teams_stats.total_wins_per_team
+    assert_equal 0, @game_teams_stats.total_wins_per_team(18)
   end
 
   def test_it_can_get_a_teams_average_win_percentage
-    assert_equal 6.25, @game_teams_stats.average_win_percentage(20)
+    assert_equal 0.0, @game_teams_stats.average_win_percentage(20)
```
Comment (Yetidancer) [inline test/game_teams_stats_test.rb:25]: to be made into mocks

File: lib/game_teams_stats.rb (hunk 2 above)
Comment (tylertomlinson) [reply lib/game_teams_stats.rb:30]: Just removed.

File: lib/stat_tracker.rb (hunk 4 above)
Comment (tylertomlinson) [reply lib/stat_tracker.rb:110]: Thanks!

File: lib/game_teams_stats.rb (hunk 1 above)
Comment (tylertomlinson) [reply lib/game_teams_stats.rb]: Brain hurts lol. Fixed now thanks

File: lib/game_teams_stats.rb (hunk 3 above)
Comment (tylertomlinson) [reply lib/game_teams_stats.rb:10]: NP

File: test/game_teams_stats_test.rb (hunk 5 above)
Comment (tylertomlinson) [reply test/game_teams_stats_test.rb:25]: Si

**Pull request merged**